src/nfa/nfa.cc

```diff
@@ -1281,31 +1281,70 @@ Nfa Mata::Nfa::minimize(
 	return algo(aut);
 }
 
-void Nfa::print_to_DOT(std::ostream &outputStream) const {
-    outputStream << "digraph finiteAutomaton {" << std::endl
+std::string Nfa::print_to_DOT() const {
+    std::stringstream output;
+    print_to_DOT(output);
+    return output.str();
+}
+
+void Nfa::print_to_DOT(std::ostream &output) const {
+    output << "digraph finiteAutomaton {" << std::endl
                  << "node [shape=circle];" << std::endl;
 
-    for (State finalState: final) {
-        outputStream << finalState << " [shape=doublecircle];" << std::endl;
+    for (State final_state: final) {
+        output << final_state << " [shape=doublecircle];" << std::endl;
     }
 
     const size_t delta_size = delta.num_of_states();
-    for (State s = 0; s != delta_size; ++s) {
-        for (const Move &t: delta[s]) {
-            outputStream << s << " -> {";
-            for (State sTo: t.targets) {
-                outputStream << sTo << " ";
+    for (State source = 0; source != delta_size; ++source) {
+        for (const Move &tran: delta[source]) {
```

Comment:
Is seems strange to use `tran` (as in, transition) if the value is a move (as in, a symbol and a set of target states). We can keep it here, but I would encourage you to use `move` if possible in such cases. It is more readable, I think.

Reply:
Fixed